### ISSUE TITLE
feat: add option to pass without tests using `trigger`

### ISSUE THREAD

**Issue by tdeekens** [checkly/checkly-cli#685]
### What problem does this feature solve?

Depending on the usage of the `trigger` command it might be expected that no tests can be triggered as none exists. A use case in which this is expected is when running Checkly across many environments from a CI pipeline dynamically while certain non crucial environments might not have any checks defined matching the `--tags` selected.

Failing if no checks are found is a very good default. For instance Jest employs this too but allows opting out with a [--passWithNoTests](https://jestjs.io/docs/cli#--passwithnotests). 

### How would you implement this feature?

Adding a `--pass-without-checks` could be an option added to the `trigger` command to opt out of failing if no matching checks are found.

**Comment by tnolet** [comment]
@tdeekens I think we can make that simpler. The `trigger` command can just pass with return code `0`. Right now we do indeed return an error. But I agree the situation is not an "error".

```bash
timnolet@MacBook-Pro-van-Tim salmon-bison % npx checkly trigger   

Unable to run checks: No matching checks were found.
timnolet@MacBook-Pro-van-Tim salmon-bison % echo $?
1
timnolet@MacBook-Pro-van-Tim salmon-bison %
```

@clample what do you think?

**Comment by tdeekens** [comment]
Would also work for me. The users is informed about the fact that there are no tests anyway.

**Comment by clample** [comment]
@tdeekens thank you for the suggestion/feedback! The CI scenario that you mention makes sense.

@tnolet good point, we can just log `No matching checks were found.` and exit with `0`. 

I checked the behavior of `checkly test`, to make sure that the two commands are consistent. It exits with `0` and a message in that case as well:
```
$ npx checkly test
Parsing your project... done
Unable to find checks to run.
$ echo $?
0
```

**Comment by tdeekens** [comment]
As a last idea: you could also just exit with 0 on CI when there are no tests in trigger and test while if not on CI you exit with 1.

**Comment by clample** [comment]
> As a last idea: you could also just exit with 0 on CI when there are no tests in trigger and test while if not on CI you exit with 1.

That's an interesting idea. I think that the different CI vs local behavior could cause some confusion for users, though. 

If any users need the failure in case of no checks being found, we could add a `--fail-without-checks` flag (similar to the `--pass-without-checks` proposed in this issue).

@tnolet what do you think?

**Comment by tdeekens** [comment]
Yeah the confusion is the downside. To be clear: I wouldn't request the flag if we would exit with 0 in any case. I don't have a strict use case to ever exit with 1.

**Comment by tnolet** [comment]
> > As a last idea: you could also just exit with 0 on CI when there are no tests in trigger and test while if not on CI you exit with 1.
> 
> 
> 
> That's an interesting idea. I think that the different CI vs local behavior could cause some confusion for users, though. 
> 
> 
> 
> If any users need the failure in case of no checks being found, we could add a `--fail-without-checks` flag (similar to the `--pass-without-checks` proposed in this issue).
> 
> 
> 
> @tnolet what do you think?

@clample let's stick to consistent behavior.
